From: Georgia Garcia <georgia.garcia@canonical.com>
Date: Wed, 8 Apr 2026 16:23:44 -0300
Subject: [PATCH] tests: check if skb mediation is enabled

skb mediation can be enabled/disabled by changing the values of
/proc/sys/kernel/apparmor_packet_mediation therefore we need to check
if it's enabled before testing.

Signed-off-by: Georgia Garcia <georgia.garcia@canonical.com>
---
 tests/regression/apparmor/net_iface.sh | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tests/regression/apparmor/net_iface.sh b/tests/regression/apparmor/net_iface.sh
index a184942..4dd9b04 100755
--- a/tests/regression/apparmor/net_iface.sh
+++ b/tests/regression/apparmor/net_iface.sh
@@ -21,6 +21,18 @@ bin=$pwd
 requires_kernel_features network_v9_skb/iface
 requires_parser_support "network interface=eth0,"
 
+skb_enabled_path="/proc/sys/kernel/apparmor_packet_mediation"
+if [ ! -e $skb_enabled_path ]; then
+	echo "$skb_enabled_path not available. Skipping tests ..."
+	exit 0
+else
+	skb_enabled=$(cat $skb_enabled_path)
+	if [ $skb_enabled -eq 0 ]; then
+		echo "$skb_enabled_path disabled. Skipping tests ..."
+		exit 0
+	fi
+fi
+
 settest net_inet_rcv
 
 sender="$bin/net_inet_snd"
